src/core/net/dns_headers.hpp

```diff
@@ -735,6 +735,17 @@ class Name
      */
     static otError CompareName(const Message &aMessage, uint16_t &aOffset, const Message &aMessage2, uint16_t aOffset2);
 
+    /**
+     * This static method tests if a DNS name ends with given domain.
+     *
+     * @param[in]  aName    The dot-separated name.
+     * @param[in]  aDomain  THe dot-separated domain.
+     *
+     * @returns  TRUE if the name matches the domain, FALSE if doesn't match.
+     *
+     */
+    static bool MatchDomain(const char *aName, const char *aDomain);
```

Comment:
I am hoping to use this method for match service instance name with service name. Can we rename it to be more general? e.g. `IsSubNameOf`.

Reply:
I think the DNS name system is built atop of the `domain` concept. For example, `_ipp._tcp.example.com.` is a subdomain but not a subname:

> Alternatively, a printer manager device could implement enough of the
   DNS protocol that it is able to answer DNS queries directly, and
   Example Co.'s main DNS server could delegate the
   "_ipp._tcp.example.com." subdomain to the printer manager device.

https://tools.ietf.org/html/rfc6763#section-10

So I think `IsSubDomainOf` is a more generic and appropriate name, if you prefer.

Reply:
I think `IsSubDomain` can be a good choice.